Create admin role

ahharris117 opened the pull request:
added role to users, added destroy method and back end authorization in restaurants api controller. Added to react: show fetch also gets current user, authenticates user as admin to show delete button, confirms if user wants to delete restaurant. Updated api controller testing to include user info in fetch.

dovakeen118 commented on app/javascript/react/components/App.js:
```diff
@@ -9,6 +9,7 @@ export const App = (props) => {
     <div>
       <BrowserRouter>
         <Switch>
+        <Route exact path='/' component={RestaurantIndexContainer} />
```
Route should be indented

dovakeen118 commented on config/routes.rb:
```diff
@@ -7,7 +7,7 @@
   namespace :api do
     namespace :v1 do
-      resources :restaurants, only: [:index, :show]
+      resources :restaurants, only: [:index, :show, :destroy, :delete]
```
only need `destroy`, that is the action defined in your controller

dovakeen118 commented on app/controllers/api/v1/restaurants_controller.rb:
```diff
@@ -1,9 +1,31 @@
+    response_body = []
+    response_body.push(user)
+    response_body.push(restaurant)
+    render json: response_body
```
instead of pushing into an array, it might be clearer to create a hash
`render json: { restaurant: restaurant, user: user }`
that way when the response gets to the react component we can call on each key directly instead of its index

ahharris117 commented:
added role to users, added destroy method and back end authorization in restaurants api controller. Added to react: show fetch also gets current user, authenticates user as admin to show delete button, confirms if user wants to delete restaurant. Updated api controller testing to include user info in fetch.

Added suggested changes.
